homeassistant/components/icloud/account.py

```diff
@@ -137,8 +150,21 @@ def update_devices(self) -> None:
             return
         except Exception as err:  # pylint: disable=broad-except
             _LOGGER.error("Unknown iCloud error: %s", err)
-            self._fetch_interval = 5
-            dispatcher_send(self.hass, SERVICE_UPDATE)
+            self._fetch_interval = 2
+            dispatcher_send(self.hass, self.signal_device_update)
+            track_point_in_utc_time(
+                self.hass,
+                self.keep_alive,
+                utcnow() + timedelta(minutes=self._fetch_interval),
+            )
+            return
+
+        iphone = api_devices.get(list(api_devices)[0])
+        _LOGGER.warning(iphone)
+        _LOGGER.warning(DEVICE_STATUS_CODES.get(list(api_devices)[0][DEVICE_STATUS]))
+        if DEVICE_STATUS_CODES.get(list(api_devices)[0][DEVICE_STATUS]) == "pending":
```

Comment:
Needs better `None` handling

Reply:
`!== 'online'`

Reply:
No because it will have side effects with sold/not used devices still in the account.